packages/block-library/src/button/editor.scss

```diff
@@ -31,6 +31,16 @@
 		opacity: 0.8;
 	}
 
+	// Make placeholder disappear on focus to make focus-state visible.
+	.block-editor-rich-text__editable[data-is-placeholder-visible="true"]:focus + .block-editor-rich-text__editable {
+		opacity: 0;
+	}
+
+	// Align cursor to left when text input is focused and placeholder is active.
+	.block-editor-rich-text__editable[data-is-placeholder-visible="true"]:focus {
```

Comment:
@nfmohit-wpmudev I think left-aligning the cursor here at all times (not just on focus) seems like a reasonable fix for [the Safari issue I described above](https://github.com/WordPress/gutenberg/pull/15058#issuecomment-484944031). 

With that change, the cursor stays active in the field after switching alignment, but it's left-aligned instead of centered all of a sudden. Since the field isn't technically focused, the placeholder text is active again (and the cursor is therefore pretty dim), but I'm not sure there's an easy way to fit that part. 

Here's a quick GIF: 

![animation](https://user-images.githubusercontent.com/1202812/56682295-42e76080-6699-11e9-91e8-1fcd31046dbf.gif)

Reply:
> @nfmohit-wpmudev I think left-aligning the cursor here at all times (not just on focus) seems like a reasonable fix for the Safari issue I described above.
I have applied this suggestion on [this commit](https://github.com/WordPress/gutenberg/pull/15058/commits/a06f391f648fec87ceb7e4c6172bc86a42041cfe).

In the case of Safari, I'm really not sure how to retain the focus styles when alignment is changed without affecting the other browsers, as the focus state itself is not retained physically. I'm looking forward to feedback and suggestions regarding this. Thank you ❤️

Reply:
Yeah, I ran into that issue too. It's a bit of an edge case, so I think this is good enough for now. 👍 Nice work!